hcp/common.go

```diff
@@ -0,0 +1,145 @@
+package hcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/stable/2021-04-30/models"
+	sdkpacker "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer/internal/registry"
+	"github.com/mitchellh/mapstructure"
+
+	registryimage "github.com/hashicorp/packer-plugin-sdk/packer/registry/image"
+)
+
+func populateIteration(
+	ctx context.Context,
+	handler HCPHandler,
+) error {
+	err := handler.Bucket().Initialize(ctx)
+	if err != nil {
+		return fmt.Errorf("HCP Packer Registry build initialization failed: %s", err)
+	}
+
+	err = handler.Bucket().PopulateIteration(ctx)
+	if err != nil {
+		return fmt.Errorf("HCP Packer Registry build initialization failed: %s", err)
```

Comment:
Indeed, copy-pasta issue here.
I guess the HCP error could be returned directly, the context will be improved upon from the callsite anyway as it will end-up wrapped in an `hcl.Diagnostic`.